src/state/graphicalItemsSlice.ts

```diff
@@ -34,15 +46,19 @@ const graphicalItemsSlice = createSlice({
     removeBar(state) {
       state.countOfBars -= 1;
     },
-    addGraphicalItemData(state, action: PayloadAction<ChartData>) {
-      state.graphicalItemData.push(action.payload);
+    addCartesianGraphicalItem(state, action: PayloadAction<CartesianGraphicalItemSettings>) {
+      state.cartesianItems.push(castDraft(action.payload));
     },
-    removeGraphicalItemData(state, action: PayloadAction<ChartData>) {
-      state.graphicalItemData = state.graphicalItemData.filter(item => item !== action.payload);
+    removeCartesianGraphicalItem(state, action: PayloadAction<CartesianGraphicalItemSettings>) {
+      const index = current(state).cartesianItems.indexOf(action.payload);
```

Comment:
why do we need `current`?

Reply:
Immer wraps objects in a Proxy. I want to remove state elements based on comparing the reference so I need a reference to the original state, not the proxy. So, current.

Redux recommendation is to use some sort of ID. But Recharts doesn't use IDs (except axes) so we need current.

Reply:
Got it, thank you!